src/amber/websockets/client_socket.cr

```diff
@@ -73,17 +73,20 @@ module Amber
         end
       end
 
-      def session
-        @session ||= @context.session
+      # Authentication and authorization can happen here
+      def on_connect : Bool
+        true
       end
 
-      def cookies
-        @cookies ||= @context.cookies
+      # On socket disconnect functionality
+      def on_disconnect; end
```

Comment:
why not `abstract def on_disconnect; end`?

Reply:
I didn't make it abstract, because its optional